core/src/test/resources/sql/sub-query.iq

```diff
@@ -5151,4 +5151,100 @@ EnumerableCalc(expr#0..2=[{inputs}], expr#3=[IS FALSE($t1)], expr#4=[null:BOOLEA
             EnumerableValues(tuples=[[{ null }, { null }]])
 !plan
 
+# [CALCITE-1583] Wrong results for query with correlated subqueries with aggregate subquery expression
+# Correlated sub-query with aggregate expression can optimized by Metadata RowCount
+select * from emp where exists (select count(deptno) from dept where dept.deptno = emp.deptno);
++-------+--------+--------+
```

Comment:
If these have been validated using another database, please add a comment.
Is there value in having the plan here?

Reply:
It has been verified in PostgreSQL. This plan can prove that this SQL has been optimized by the Metadata RowCount method.

Reply:
could you add these as comments?
is the optimization relevant for the issue being resolved?

Reply:
Since the discussion in Jira involves this SQL, I have included it in the test cases. The comments have already been added.